File: Backend/src/main/java/pe/com/model/Recibo.java
package pe.com.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "recibo")
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Recibo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column
    private String year;

    @Column
    private String month;

    @Column
    private String particulars;

    @Column
    private String total;

    @Column
    private String comments;

    @Column
    private String fecCreacion;

    @Column
    private String usuCreacion;

    @Column
    private String fecModifica;

    @Column
    private String usuModifica;

    @ManyToOne
    @JoinColumn(name = "id_departamento", nullable=false)
    private Departamento idDepartamento;

    @ManyToOne
    @JoinColumn(name = "id_estado_recibo", nullable=false)
    private EstadoRecibo idEstadoRecibo;
}

File: Frontend/src/app/pages/model/recibo.ts
import { Departamento } from "./departamento";
import { EstadoRecibo } from "./estadoRecibo";

export interface Recibo {
    id:number,
    year:string,
    month:string,
    particulars:string,
    total:string,
    comments:string,
    fecCreacion:string,
    usuCreacion:string,
    fecModifica:string,
    usuModifica:string,
    idDepartamento:Departamento,
    idEstadoRecibo:EstadoRecibo
}